jenkins/docker-ecr/Jenkinsfile

```diff
@@ -0,0 +1,152 @@
+                            sh '''
+                                set -ex
+                                echo Login to $CREDENTIAL_ID
+                                docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD && eval $DOCKER_BUILD_SCRIPT_WITH_COMMANDS
+                            '''
+                        }
+                    }
+                }
+            }
+            post() {
+                always {
+                    script {
+                        postCleanup()
+                        sh "docker logout"
+                    }
+                }
+            }
+        }
+        stage('docker-build-ecr') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-Ubuntu2004-X64-m52xlarge-Docker-Builder'
+                    image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+                    args '-u root'
+                }
+            }
+            steps {
+                script {
+                    install_gcrane()
+                    if (env.ENVIRONMENT == 'production')
+                        withAWS(role: 'abhng-ecr-role', roleAccount: "${ABHNG_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-session') {
+                            sh '''
+                                docker logout
+                                aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/a5n0l3m1
+                                gcrane cp public.ecr.aws/a5n0l3m1/opensearch:${SOURCE_IMAGE_TAG_IN_STAGING} public.ecr.aws/a5n0l3m1/opensearch:${DESTINATION_IMAGE_TAG_IN_PROD}
+                            '''
+                        } else {
+                        withAWS(role: 'abhng-ecr-role', roleAccount: "${ABHNG_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-session') {
+                            sh '''
+                                docker logout
+                                aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/a5n0l3m1
+                                gcrane cp opensearchstaging/opensearch:latest public.ecr.aws/a5n0l3m1/opensearch-copy:latest
+                            '''
+                        }
+                    }
+                }
+            }
+            post() {
+                always {
+                    script {
+                        postCleanup()
+                        sh "docker logout"
+                    }
+                }
+            }
+        }
+    }
+}
+
+void install_gcrane(){
+    sh'''
+        curl -L https://github.com/google/go-containerregistry/releases/latest/download/go-containerregistry_Linux_x86_64.tar.gz \\
+            -o go-containerregistry.tar.gz
+        tar -zxvf go-containerregistry.tar.gz
+        chmod +x gcrane
+        mv gcrane /usr/local/bin/
+      '''
```

Comment:
This should be directly as part of the ci-runner image.

Reply:
Instead of continuing to add onto a 'central' image we should create a separate images for each type of task.

Reply:
I agree with @peternied since this is very specific to this task only. But would we not end up creating a lot of images over a period of time? We need to come up with some plan to follow. Thoughts?

Reply:
I think that we shouldn't update the a central CI image.

However, running curl/install each execution is can running this job with an additional dependency that can fail.  I am not worried about the number of images so long as how they are build has the docker file and a way to search them.  

https://github.com/opensearch-project/opensearch-build/issues/1207 would be a good place for more discussion